src/type/order.type.ts

```diff
@@ -31,12 +38,33 @@ export interface OrderSchemaModel {
   userId: mongoose.Types.ObjectId;
   activityId: mongoose.Types.ObjectId;
   ticketId: mongoose.Types.ObjectId;
-  name: string;
-  email: string;
-  phone: string;
-  startDateTime: Date;
-  endDateTime: Date;
+  orderContact: OrderContact;
   payment: OrderPayment;
   orderStatus: OrderStatusEnum;
   serialNumber: string;
 }
+
+export type CreateOrderParams = {
+  userId: mongoose.Types.ObjectId;
+  requestBody: Pick<
+    OrderSchemaModel,
+    'activityId' | 'ticketId' | 'orderContact' | 'payment'
+  >;
+};
+
+export type CreateOrderCredentials = Pick<
+  OrderSchemaModel,
+  'orderContact' | 'payment'
+>;
+
+export type GetOrdersParams = {
+  userId?: mongoose.Types.ObjectId;
+  page?: number;
+  limit?: number;
+  sort?: SortType;
+};
+
+export type CancelOrderParams = {
+  userId?: mongoose.Types.ObjectId;
```

Comment:
why userId use "?:"

Reply:
userId may be undefined from req.user, think more concise here XD I would change to align the same as other user type